### PR TITLE
Style changes

### PR DESCRIPTION
**Pull request by dkfellows**
More features from checkstyle so that we can have a less strict style policy for test code (e.g., to allow magic numbers). But we do add in one key piece of strictness (actually several, but the others are no big deal): fields (NB: not local variables) must have a blank line between them. To see what I mean, look at this:
```java
class Foo {
    // bad
    int x;
    int y;
}
```
```java
class Foo {
    // also bad
    int x, y;
}
```
```java
class Foo {
    // good
    int x;

    int y;
}
```

And added a bunch more tests.

----

Most of the commits on this branch are due to fixing the style due to the above change.

### CODE REVIEW

**Comment by dkfellows** [comment]
Sorry about the sheer size of this one. It touches a lot of files (in a trivial way)

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/spalloc/SpallocJob.java [hunk 1]
```diff
@@ -218,6 +231,60 @@ public SpallocJob(CreateJob builder)
 				builder);
 	}
 
+	/**
```
Comment (Christian-B) [inline SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/spalloc/SpallocJob.java:234]: Did you intend for this to be included in this PR?

File: SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/spalloc/TestClient.java [hunk 2]
```diff
@@ -274,19 +284,267 @@ void testWaitForNotification() throws Exception {
 	}
 
 	@Test
-	void testCommandsAsMethods() throws Exception {
+	@SuppressWarnings("deprecation")
+	void testCommandCreateJob() throws Exception {
```
Comment (Christian-B) [inline SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/spalloc/TestClient.java:288]: Again looks fine but it it intentional to be in this PR

File: SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/spalloc/TestMockClient.java [hunk 3]
```diff
@@ -49,158 +49,172 @@
...
- * Tests the Spalloc Client ideally using an actual connection
- *      but with a backup of canned replies if the connection fails/ timesout.
+ * Tests the Spalloc Client ideally using an actual connection but with a backup
+ * of canned replies if the connection fails/times out.
  *
  * @author Christian
  */
+@SuppressWarnings("unused")
 public class TestMockClient {
+	private static final int SECOND = 1000;
```
Comment (Christian-B) [inline SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/spalloc/TestMockClient.java:59]: Another file with code changes

**Review by Christian-B — APPROVED** [review]
Assuming that the SpallocJob changes are not accidental this is approved with our without the Spalloc Job changes

File: SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/spalloc/SpallocJob.java (hunk 1 above)
Comment (dkfellows) [reply SpiNNaker-comms/src/main/java/uk/ac/manchester/spinnaker/spalloc/SpallocJob.java:234]: I noticed that the API wasn't connected enough when adjusting the tests.

I perhaps ought to make the new tests and this bit their own PR.

**Comment by dkfellows** [comment]
Obviously my attention wasn't too great in the immediate post-vaccination state.

File: SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/spalloc/TestMockClient.java (hunk 3 above)
Comment (dkfellows) [reply SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/spalloc/TestMockClient.java:59]: Hmm, this one originally came from a branch that was disallowing magic numbers in unit tests. I hated that aspect of style enforcement (after playing with it for a while).

**Pull request merged**